src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
   // write code here;
+  return items.reduce((grops, item) => {
+    const keyValue = String(item[key]);
+
+    if (!grops[keyValue]) {
```

Comment:
The typo in the variable name `grops` appears here as well. It should be corrected to `groups`.